src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -421,6 +422,17 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
                             string.Format(Core.Resources.RevIncludeIterateTargetTypeNotSpecified, p.query));
                     }
 
+                    if (expression.TargetResourceType != null &&
+                       string.IsNullOrWhiteSpace(expression.TargetResourceType))
+                    {
+                        throw Error.Format($"Invalid TargetResourceType : value cannot be empty");
+                    }
+
+                    if (expression.TargetResourceType != null && !ModelInfoProvider.IsKnownResource(expression.TargetResourceType))
```

Comment:
nit: Should we remove this check from here as its already handled above
expression.TargetResourceType != null

Reply:
First condition is to handle empty or white spaces.
Second condition is to validate if given Target resource type is invalid.